### PR TITLE
Kaito fed creds and identity changes

### PR DESCRIPTION
**Pull request by hsubramanianaks**
This PR has the following work for Kaito - Thanks to @Tatsinnit for his kind contribution and awesome work.

1. Role assignments and Federated credentials via azure/identity, azure/arm-authorization, azure/arm-msi sdk's,  more details here - https://learn.microsoft.com/en-us/azure/aks/ai-toolchain-operator#export-environment-variables-1
2. Retry logic in case of delay in preview feature registration 
3. some minor ux changes to include progress spinner.
Screenshot 2024-09-17 193153
Screenshot 2024-09-17 193242

### CODE REVIEW

File: src/commands/aksKaito/aksKaito.ts [hunk 1]
```diff
@@ -17,12 +19,33 @@ export default async function aksKaito(_context: IActionContext, target: unknown
         return;
     }
 
+    if (!kubectl.available) {
```
Comment (ReinierCC) [inline src/commands/aksKaito/aksKaito.ts:22]: There's a check for kubectl below. Would the kubectl state have changed?

File: src/panels/KaitoPanel.ts [hunk 2]
```diff
@@ -117,45 +139,57 @@ export class KaitoPanelDataProvider implements PanelDataProvider<"kaito"> {
...
+            },
+        );
 
-        if (featureRegistrationPoller.properties?.state !== "Registered") {
-            webview.postKaitoInstallProgressUpdate({
-                operationDescription: "Installing Kaito",
-                event: 3,
-                errorMessage: "Failed to register feature",
-                models: [],
+        if (getFeatureClientRegisterState.properties?.state !== "Registered") {
```
Comment (Tatsinnit) [inline src/panels/KaitoPanel.ts:150]: 💡 Self note: all this need some refactor which we can do in the final drop - basically , bunch of Private functs and `longRunning` job for better loop back.

File: src/panels/KaitoPanel.ts [hunk 3]
```diff
@@ -117,45 +139,57 @@ export class KaitoPanelDataProvider implements PanelDataProvider<"kaito"> {
...
+            const featureRegistrationPoller = await longRunning(`Registering the AIToolchainOperator.`, () => {
+                return this.featureClient.features.register(
+                    "Microsoft.ContainerService",
+                    "AIToolchainOperatorPreview",
+                    {},
+                );
             });
-            return;
+
+            if (featureRegistrationPoller.properties?.state !== "Registered") {
```
Comment (Tatsinnit) [inline src/panels/KaitoPanel.ts:160]: 💡 Same goes for this: https://github.com/Azure/vscode-aks-tools/pull/914/files#r1766176538

**Review by tejhan — COMMENTED** [review]
Looks good to me 👍

File: src/commands/aksKaito/aksKaito.ts (hunk 1 above)
Comment (hsubramanianaks) [reply src/commands/aksKaito/aksKaito.ts:22]: This is just a check to make sure kubectl api via k8s extension is available. When you say below is it about creating temp file for kubeconfig? @ReinierCC

Comment (hsubramanianaks) [reply src/commands/aksKaito/aksKaito.ts]: Removed the duplicate lines Thanks @ReinierCC

**Review by ReinierCC — APPROVED** [review]
LGTM, Great work

**Review by tejhan — COMMENTED** [review]
Looks good ! 👍

**Review by tejhan — APPROVED** [review]
Looks good!👍

**Comment by ivelisseca** [comment]
Awesome, the Kaito installed status now shows the progress icon!

**Review by ivelisseca — COMMENTED** [review]
Awesome, the Kaito installed status now shows the progress icon!

**Comment by qpetraroia** [comment]
Sorry for the late review. @hsubramanianaks, this looks great. Thank you!

**Pull request merged**